.github/workflows/release.yml

```
@@ -0,0 +1,79 @@
+name: Release Packages
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      dryRun:
+        type: choice
+        description: 'Dry run'
+        required: true
+        default: 'false'
+        options:
+          - 'true'
+          - 'false'
+
+env:
+  NX_NON_NATIVE_HASHER: true
+
+jobs:
+  dry-run-release:
+    if: |
+      github.repository == 'code-pushup/cli' && (
+        github.event_name == 'pull_request' ||
+        (github.event_name == 'workflow_dispatch' && github.event.inputs.dryRun == 'true')
+      )
+    name: Dry run release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: npm
+      - name: Install dependencies
+        run: npm ci
+      - name: Execute version target on all affected packages and release them
+        run: npx nx affected --base=latest --target=version --dryRun
+  release:
+    if: |
+      github.repository == 'code-pushup/cli' && (
+        (github.event_name == 'workflow_dispatch' && github.event.inputs.dryRun == 'false') ||
+        (github.event_name == 'push')
+      )
+    name: Release packages
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: npm
+      - name: Setup Git
+        run: |
+          git config user.name "GitHub Bot"
+          git config user.email "gituser@example.com"
+      - name: Install dependencies
+        run: npm ci
+      - name: Execute version target on all affected packages and release them
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: npx nx affected --base=latest --target=version
```

I suggest we exclude `plugin-lighthouse` and `nx-plugin` for now, since it will still be some time before we're ready to release those.

```suggestion
        # un-exclude packages when they're ready for public release
        run: npx nx affected --base=latest --target=version --exclude=plugin-lighthouse,nx-plugin
```